src/jdk.incubator.jextract/share/classes/jdk/internal/clang/TranslationUnit.java

```diff
@@ -62,9 +62,8 @@ public Cursor getCursor() {
 
     public final void save(Path path) throws TranslationUnitSaveException {
         try (MemorySegment pathStr = Utils.toNativeString(path.toAbsolutePath().toString())) {
-            int res = Index_h.clang_saveTranslationUnit(tu,
-                    pathStr.baseAddress(), 0);
-            if (res != 0) {
+            SaveError res = SaveError.valueOf(Index_h.clang_saveTranslationUnit(tu, pathStr.baseAddress(), 0));
+            if (res != SaveError.None) {
                 throw new TranslationUnitSaveException(path);
```

Comment:
Would it be useful to store the SaveError kind in TranslationUnitSaveException?

Reply:
Yes, good idea.